src/events/mod.ts

```diff
@@ -0,0 +1,62 @@
+import { SlackManifest } from "../manifest/mod.ts";
+import { ManifestCustomEventSchema } from "../manifest/manifest_schema.ts";
+import {
+  CustomEventDefinition,
+  DefineEventSignature,
+  ICustomEvent,
+} from "./types.ts";
+
+export const DefineEvent: DefineEventSignature = <
+  Def extends CustomEventDefinition,
+>(
+  definition: Def,
+) => {
+  return new CustomEvent(definition);
+};
+
+export class CustomEvent<Def extends CustomEventDefinition>
+  implements ICustomEvent {
+  public id: string;
```

Comment:
Should we just call this `name`?

Reply:
I chose `id` because it matches Custom Types and events is pretty much the same. That being said, if we went with `name` it will match Datastores (and the property that's being set). I don't have a strong preference here, open to changing if anyone else does!